apps/site/lib/site_web/controllers/schedule/line/diagram_format.ex

```diff
@@ -146,8 +146,10 @@ defmodule SiteWeb.ScheduleController.Line.DiagramFormat do
 
         chunk ->
           last_stop = List.last(chunk)
+          last_stop_fx = effects_for_stop(last_stop)
+          has_same_effect = Enum.any?(last_stop_fx, fn effect -> effect in stop_fx end)
 
-          if stop_fx == effects_for_stop(last_stop) do
+          if has_same_effect do
```

Comment:
I think a way of kind of cleaning up at least a couple of the nested if/case statements here is something like this. Lets you evaluate both checks w/ the same case and saves some indents.

```elixir
case {last_stop_fx -- stop_fx, stop_fx} do
  {^last_stop_fx, []} -> {:cont, chunk, []}
  {^last_stop_fx, _} -> {:cont, chunk, [stop]}
  _ -> {:cont, chunk ++ [stop]}
end
```

```elixir
iex(15)> [1, 2, 3] -- [4, 5, 6]
[1, 2, 3]
iex(16)> [1, 2, 3] -- [4, 5, 6, 1]
[2, 3]
```